edge-apps/weather/static/js/main.js

```diff
@@ -0,0 +1,315 @@
+    getDataFromIndexedDB(db, (data) => {
+      const { list } = data
+      const currentIndex = findCurrentWeatherItem(list)
+
+      const { dt, weather, main: { temp } } = list[currentIndex]
+
+      if (Array.isArray(weather) && weather.length > 0) {
+        const { id, description } = weather[0]
+        const { icon, bg } = getWeatherImagesById(id, dt)
+        if (id !== currentWeatherId) {
+          loadBackground(bg)
+        }
+
+        updateCurrentWeather(icon, description, temp)
+        currentWeatherId = id
+      }
+
+      const weatherListContainer = document.querySelector('#weather-item-list')
+      const frag = document.createDocumentFragment()
+      const windowSize = 5
+      const currentWindow = list.slice(currentIndex, currentIndex <= windowSize - 1 ? currentIndex + windowSize : list.length - 1)
+      currentWindow.forEach((item, index) => {
+        const { dt, main: { temp }, weather } = item
+
+        const { icon } = getWeatherImagesById(weather[0]?.id, dt)
+        const dateTime = moment.unix(dt).utcOffset(tz)
+
+        const dummyNode = document.querySelector('.dummy-node')
+        const node = dummyNode.cloneNode(true)
+        node.classList.remove('dummy-node')
+        node.querySelector('.item-temp').innerText = getTemp(temp)
+        node.querySelector('.item-icon').setAttribute('src', icons[icon])
+        node.querySelector('.item-time').innerText = index === 0 ? 'Current' : formatTime(dateTime)
+
+        frag.appendChild(node)
+      })
+
+      weatherListContainer.innerHTML = ''
+      weatherListContainer.appendChild(frag)
+    })
+    weatherTimer = setTimeout(() => updateWeather(), weatherDataUpdateInterval)
+  }
+
+  const updateData = () => {
+    getDataFromIndexedDB(db, (data) => {
+      const { name, country, timezone } = data
+      tempScale = countriesUsingFahrenheit.includes(country) ? 'F' : 'C'
+      updateLocation(name)
+      tz = setTimeZone(timezone)
+      initDateTime()
+    })
+
+    updateWeather()
+  }
+
+  const cacheWeatherData = (data) => {
+    const { city: { name, country, timezone }, list } = data
+    const transaction = db.transaction(['weatherStore'], 'readwrite')
+    const objectStore = transaction.objectStore('weatherStore')
+    const addRequest = objectStore.add({ name, country, timezone, list })
+
+    addRequest.addEventListener('success', () => {
+      console.log('Weather data added to the store')
+    })
+
+    transaction.addEventListener('complete', () => {
+      console.log('Transaction completed: database modification finished')
+    })
+
+    transaction.addEventListener('error', () => {
+      console.error('Transaction not opened due to error')
+    })
+  }
+
+  const fetchWeather = async () => {
+    try {
+      const endpointUrl = `https://api.openweathermap.org/data/2.5/forecast`
+      const apiKey = screenly.settings.openweathermap_api_key
+      const queryParams = `lat=${lat}&lon=${lng}&units=metric&cnt=10&appid=${apiKey}`
```

Comment:
Maybe it's easier to just pass the right unit here (`matric` is hard coded) than to do Fahrenheit conversion.

Reply:
I'll address this comment on another issue or pull request. I'll keep you posted.